qiskit/extensions/standard/multi_control_rotation_gates.py

```diff
@@ -0,0 +1,239 @@
+# -*- coding: utf-8 -*-
+
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2018, 2019.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""
+Multiple-Control U3 gate. Not using ancillary qubits.
+"""
+
+import logging
+from math import pi
+from sympy.combinatorics.graycode import GrayCode
```

Comment:
sympy is on its way to being removed as a dependency (#3156). Is this strictly necessary or can we replace it with another GrayCode impelementation?